src/Domain.LinnApps/Reports/ImportBookReportService.cs

```diff
@@ -0,0 +1,122 @@
+﻿namespace Linn.Stores.Domain.LinnApps.Reports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Common.Reporting.Layouts;
+    using Linn.Common.Reporting.Models;
+    using Linn.Stores.Domain.LinnApps.ImportBooks;
+
+    public class ImportBookReportService : IImportBookReportService
+    {
+        private readonly IRepository<ImportBook, int> impbookRepository;
+
+        private readonly IReportingHelper reportingHelper;
+
+        private readonly int IprCpcNumberId = 13;
+
+        public ImportBookReportService(IRepository<ImportBook, int> impbookRepository, IReportingHelper reportingHelper)
+        {
+            this.impbookRepository = impbookRepository;
+            this.reportingHelper = reportingHelper;
+        }
+
+        public ResultsModel GetIPRReport(DateTime from, DateTime to)
+        {
+            var iprImpbooks = this.impbookRepository.FilterBy(
+                x =>
+
+                    // x.IprCpcNumber.HasValue && x.IprCpcNumber.Value == 13 &&
+                    // todo check if I can just use that ^, not sure it'll be populated reliably 
+                    x.OrderDetails.Any(z => z.CpcNumber.HasValue && z.CpcNumber.Value == this.IprCpcNumberId)
```

Comment:
as discussed before 13 might work but not necessarily be future proof.  it gets us going. The most probably mistake that will happen is Rhona or someone will pick the CPC number that is close but wrong (I can see 2 this year using the extra 0 version) however all my processing will also only really work with 13

Reply:
Yeah, we could pull all of the cpc numbers in easily enough but even then, we either use the right one with id 13, or include the other similar ones and maybe get dodgy results! Maybe I can put an 'are you sure this is the right cpc number/ is this IPR?' popup in the new form to hopefully stop the wrong one being used

Reply:
i would just leave it is until we get the report working.  it is a future worry that doesn't stop you doing what you need to do